aquacomputer_d5next.c

```diff
@@ -567,16 +580,33 @@ static int aqc_set_ctrl_val(struct aqc_data *priv, int offset, long val, size_t
 	if (ret < 0)
 		goto unlock_and_return;
 
-	switch (size) {
-	case 16:
-		put_unaligned_be16((u16)val, priv->buffer + offset);
-		break;
-	case 8:
-		priv->buffer[offset] = (u8)val;
-		break;
-	default:
-		ret = -EINVAL;
+	ret = aqc_set_buffer_val(priv->buffer, offset, val, size);
+	if (ret < 0)
 		goto unlock_and_return;
+
+	ret = aqc_send_ctrl_data(priv);
+
+unlock_and_return:
+	mutex_unlock(&priv->mutex);
+	return ret;
+}
+
+/* Refreshes the control buffer, updates values at offsets and writes buffer to device */
+static int aqc_set_ctrl_vals(struct aqc_data *priv, int offset[], long val[],
```

Comment:
As far as I can see `aqc_set_ctrl_val()` can be refactored to use this with size of 1? And I'd name the three array parameters in plural, they hold multiple values after all.